qiskit_experiments/base_analysis.py

```diff
@@ -66,11 +68,11 @@ def run(
                      supported options.
 
         Returns:
-            List[AnalysisResult]: the output for analysis that produces
-                                  multiple results.
+            List[DbAnalysisResultV1]: the output for analysis that produces
```

Comment:
Does the versioning need to be shown here (and other docs that link to DbAnalysisResultV1) or can it just be `DbAnalysisResult` ?

Reply:
Furthermore, can we avoid the handling of the DB AnalysisResult here and just have the `_run_analysis` method return a list of `dict/AnalysisResultData`? Looking at the T1 Analysis it seems that the code added there to convert to a db result should be handled elsewhere.

Looking at the code for DdAnalysisResult it looks like it would be good to define a `AnalysisResultData` dict subclass for the data for both type hints (like @nkanazawa1989 wanted for curve analysis data), and to define a custom repr, so this code can be moved out of `DbAnalysisResultV1.__str__` in line 377. This would also mean we can move the `__keys_not_shown__` attribute to the AnalysisResultData class.

Reply:
+1